src/hooks/useForm.ts

```diff
@@ -0,0 +1,84 @@
+import { ChangeEvent, useEffect, useState } from 'react'
+import { emailRegex, nameRegex, requiredRegex } from '../texts/errors/formErrors'
+
+// input data types supported
+export type typeTypes = 'text' | 'name' | 'email'
+
+// taken and modified from https://learnetto.com/blog/react-form-validation
+const useForm = (defaultValues: Record<string, string>) => {
+  // Form values
```

Comment:
some comments are "you-dont-say". Good code is self-documenting, which in this case yours is. Comments add on unnecessary clutter.

Reply:
refactored